src/config-schemas/openmrs-esm-patient-registration-schema.ts

```diff
@@ -29,16 +47,24 @@ export const esmPatientRegistrationSchema = {
             label: 'phoneNumber',
             uuid: '14d4f066-15f5-102d-96e4-000c29c2a5d7',
             placeholder: 'phoneNumberPlaceHolder',
-            validation: { required: true, matches: '^[0-9]*$' },
+            validation: {
+              required: true,
+              matches: '^[0-9]*$',
+            },
           },
         ],
       },
     ],
   },
   links: {
-    submitButton: {
-      default: '${openmrsSpaBase}/patient/${patientUuid}/chart',
-      validators: [validators.isUrlWithTemplateParameters(['patientUuid'])],
+    _type: Type.Object,
+    _default: {
+      _elements: {
+        submitButton: {
+          _type: Type.String,
+          _default: '${openmrsSpaBase}/patient/${patientUuid}/chart',
+        },
+      },
```

Comment:
You should never have schema keywords nested in `_default`. `_default` should just be, literally, the default value. And use of `Type.Object` should be very rare.

```suggestion
    submitButton: {
      _type: Type.String,
      _default: '${openmrsSpaBase}/patient/${patientUuid}/chart'
    },
```

Reply:
I tried this

```typescript
links: {
    _type: Type.Object,
    _elements: {
         _submitButton: {
             _type: Type.String
         }
     },
     _default: {
         submitButton: '${openmrsSpaBase}/patient/${patientUuid}/chart'
     }
}
```

In the browser there's an error that says the schema is bad that it lacks a default. It only worked when I when with the current approach. What is the way forward for this?

Reply:
Just use the code I suggested above?

Reply:
I didn't understand it. I thought you just commented on my code, it's now I'm seeing that it's a code suggestion. Did you remove the top level `links` element?